### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Hombre2014**
All the requirements from the test phase of the project have been implemented as follow:
- [x] Set up React Testing Library.
  - [x] Follow the instructions in the official [documentation](https://testing-library.com/docs/react-testing-library/intro/).
  - [x] You don't need to setup Jest (it's included in Create React App).
- [x] Write unit tests for the files `operate.js` and `calculate.js` using Jest.
- [x] Create unit tests for all React components:
  - [x] Use Jest snapshots to test the components.
  - [x] Use React Testing Library to simulate user interaction.
- [x] Run your full tests suite using Jest. All tests should pass.
- [x] You need to use pair-programming for this project.

### CODE REVIEW

File: src/__test__/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import operate from '../logic/operate';
```
Comment (petudeveloper) [inline src/__test__/operate.test.js:1]: On the operate tests you are missing to test the `%` operation and the `÷` by `0`. Please add the necessary code to cover all the functionalities & scenarios.

File: src/__test__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
```
Comment (petudeveloper) [inline src/__test__/calculate.test.js:1]: You did a great job testing some functionalities and scenarios of the `calculate.js` module, but you are missing the following functionalities and scenarios: `AC`, `.`, `+/-`, `User pressed an operation after pressing '='`, `User pressed an operation button and there is an existing operation`. Please add the necessary code to cover all the functionalities & scenarios.

**Review by petudeveloper — CHANGES_REQUESTED** [review]
Hi @Hombre2014, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

To highlight :clap: :clap: :
- Highly organized/clean code.
- Excellent communication with the reviewer  (your message on the pull request gives the necessary info and is very well explained). 
- The README file is great.

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/__test__/calculate.test.js (hunk 2 above)
Comment (Hombre2014) [reply src/__test__/calculate.test.js:1]: Thanks for your review. We have a test case for pressing the button `AC`, please see the file [userInteractions.test.js](https://github.com/Hombre2014/math-magicians/blob/test/src/__test__/userInteractions.test.js). We will add the rest of the test scenarios you have mentioned in our next pull request.

**Review by julie-ify — APPROVED** [review]
## Hi @Hombre2014,

# Highlights
- Good job implementing the required changes from the previous reviewer ✔️ 
- All tests are passing ✔️ 

# Status: Approved ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 

Approved emoji

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**